### DISCUSSION TITLE
httpx uses proxy unexpectedly on windows

### DISCUSSION THREAD

**Discussion by amchii** [encode/httpx#1513] [Potential Issue]
### Checklist

- [x] The bug is reproducible against the latest release and/or `master`.
- [x] There are no similar issues or pull requests to fix it yet.

### Describe the bug

I set a proxy globally on `127.0.0.1:10909` but except for `localhost`/`127.0.0.1`(Windows Setting->Network->Proxy ),  then I found `httpx.get('http://127.0.0.1:8000')` will use the proxy(unexpected), but chrome and `requests.get('http://127.0.0.1:8000')` accessed it directly(expected).

### To reproduce

1. Set a proxy globally but except for `localhost`\\`127.0.0.1`,
2. Exec `httpx.get('http://127.0.0.1:8000')` .

### Expected behavior

`httpx.get` accesses `http://127.0.0.1:8000` without proxy.

### Actual behavior

`httpx.get` accesses `http://127.0.0.1:8000` with proxy.

### Environment

- OS: Windows 10
- Python version: 3.7.6
- HTTPX version: 0.17.0
- Async environment: no
- HTTP proxy: no
- Custom certificates: no

**Comment by lovelydinosaur** [comment]
We might need a bit of compare-and-contrast between what we do for determining environment proxies on `httpx` vs. what `requests` does.

Our case... https://github.com/encode/httpx/blob/e05a5372eb6172287458b37447c30f650047e1b8/httpx/_utils.py#L287
Theirs... https://github.com/psf/requests/blob/589c4547338b592b1fb77c65663d8aa6fbb7e38b/requests/utils.py#L770

**Reply by amchii** [reply]
They check the registry for the `internetSettings` and match the `proxyOverride` https://github.com/psf/requests/blob/8c211a96cdbe9fe320d63d9e1ae15c5c07e179f8/requests/utils.py#L64

**Reply by lovelydinosaur** [reply]
Okay, so pretty sure we're going to want to escalate this one. Either so that we're documenting that we *only* use environment variables for our proxy env setup (so that we're more consistent cross-platform) *or* (more likely?) to add support for windows registry proxy settings.

**Reply by amchii** [reply]
Thanks!

And macosx: https://github.com/python/cpython/blob/3ba3d513b1e3c63d09cb798b982a9e6c369cea4c/Lib/urllib/request.py#L2638.
`requests`  checks proxy-bypass setting for every certain request url, they can easily do this by calling `urllib.request.proxy_pass(host)`. But `httpx` use `httpx.Client._mounts` to save the proxy-pattern when initialization for all request, with this approach, I think it's a little hard to mount the proxy-pass pattern , especially ip like '10.0/16'.

**Reply by lovelydinosaur** [reply]
Escalated as #1536
